### PR TITLE
[custom channels]: Fix bandwidth manager and other bugs

### PR DESCRIPTION
**Pull request by guggero**
Depends on https://github.com/lightningnetwork/lnd/pull/9333.

Fixes the issue that would lead to a force close if the channel bandwidth of a target channel during a forward event isn't sufficient.

Also fixes https://github.com/lightninglabs/taproot-assets/issues/1234 (drive-by fix) and another edge case found during integration tests.

### CODE REVIEW

File: tapchannel/commitment.go [hunk 1]
```diff
@@ -363,20 +363,24 @@ func processAddEntry(htlc *DecodedDescriptor, ourBalance, theirBalance uint64,
...
 		"theirBalance=%d, ourAssetBalance=%d, theirAssetBalance=%d",
 		whoseCommit, ourBalance, theirBalance, ourAssetBalance,
 		theirAssetBalance)
 
 	var (
-		numHTLCs int64
-		feePerKw = view.FeePerKw
+		// When determining whether we need an anchor output, we need to
+		// count the non-asset HTLCs as well.
+		numHTLCs = uint64(len(nonAssetView.OurUpdates) +
```
Comment (GeorgeTsagk) [inline tapchannel/commitment.go:379]: would this lead to stuck force-closes? (covered in the [LitD itest](https://github.com/lightninglabs/lightning-terminal/blob/98c9819b81d03f926f72656d9c0fb33c010e933e/itest/litd_custom_channels_test.go#L3533-L3537)?)

trying to understand if we ever had coverage for this

**Review by GeorgeTsagk — APPROVED** [review]
LGTM! :goat:

**Comment by GeorgeTsagk** [comment]
when we do "breaking version bumps" I wonder how can we get the [`Lit Tests`](https://github.com/lightninglabs/taproot-assets/actions/runs/12162324676/job/33918992171?pr=1236) to run correctly and show :green_circle: instead of compilation errors

File: tapchannel/commitment.go (hunk 1 above)
Comment (guggero) [reply tapchannel/commitment.go:379]: No, the channel will not force close (at least from my observations), but the HTLC will be rejected. This is covered in the new `litd` itest, where we pay sats over an asset channel.

**Comment by guggero** [comment]
> when we do "breaking version bumps" I wonder how can we get the [`Lit Tests`](https://github.com/lightninglabs/taproot-assets/actions/runs/12162324676/job/33918992171?pr=1236) to run correctly and show 🟢 instead of compilation errors
> 
> Could maybe pin the LitD commit that we want to use to run the itests, then CI would use that

Yeah, it's not optimal. We could point the litd itest to an unmerged commit, but IMO that's also not great, since we need to merge `tapd` first, then update `litd` which leads to a new commit...

So I think the best strategy is just to always follow up a `tapd` change that doesn't pass the LiT itests with a PR that shows it will eventually pass.

File: tapchannel/commitment.go (hunk 1 above)
Comment (Roasbeef) [reply tapchannel/commitment.go:379]: Won't this result in us _over counting_? The non asset view will still have an manifest HTLC for a given asset HTLC. I think it's enough to just have this combined sum here, and drop the `numHTLCs` increment in the inner loop.

**Review by Roasbeef — CHANGES_REQUESTED** [review]

File: tapchannel/commitment.go (hunk 1 above)
Comment (guggero) [reply tapchannel/commitment.go:379]: No, the non-asset view and the asset view are fully distinct. And even if we were over counting, we only use `numHTLCs > 0` as a condition for determining whether we need a commitment anchor output.

Comment (guggero) [reply tapchannel/commitment.go]: I thought about this again and my initial approach wasn't 100% correct. It is correct that the `assetView` and
`nonAssetView` are non-overlapping. But we need to replicate this behavior:
https://github.com/lightningnetwork/lnd/blob/ab7aae07085ec244f7888e990e8d8cc6ae6dfa83/lnwallet/commitment.go#L706

So we also need to check for dust, which I now added.

**Review by GeorgeTsagk — APPROVED** [review]
LGTM pt.2 :star2:

**Review by ffranr — APPROVED** [review]
